aws/resource_aws_emr_managed_scaling_policy.go

```diff
@@ -0,0 +1,147 @@
+package aws
+
+import (
+	"log"
+	// "time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/emr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceAwsEMRManagedScalingPolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsEMRManagedScalingPolicyCreate,
+		Read:   resourceAwsEMRManagedScalingPolicyRead,
+		Delete: resourceAwsEMRManagedScalingPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"compute_limits": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				Elem:     computeLimitsSchema(),
```

Comment:
Except in extreme cases, we currently prefer schema declarations to be inline so there is no indirection to find the information and to align with potential future auto-generation of the schema code.

Reply:
AWS has this nested structure in the API
https://github.com/aws/aws-sdk-go/blob/master/service/emr/api.go#L9376-L9384
https://docs.aws.amazon.com/sdk-for-go/api/service/emr/#ManagedScalingPolicy
https://docs.aws.amazon.com/sdk-for-go/api/service/emr/#ComputeLimits

Reply:
I'm referring to the addition of the `computeLimitsSchema` function; it can (and should!) follow the API structure.

Reply:
sorry, I still not get your point.
In AWS API, the ManagedScalingPolicy structure has a member with type ComputeLimits structure. That's why I put computeLimitsSchema inside resourceAwsEMRManagedScalingPolicy here.

Reply:
Sorry I'm not explaining clear enough. Please take the `return` value of the `computeLimitsSchema` function and move its contents to this line and remove the "extra" `computeLimitsSchema` function declaration. Other provider resources do not split up the schema into multiple Go functions, except in rare cases, and the preference is to not split up the schema in the Go code.